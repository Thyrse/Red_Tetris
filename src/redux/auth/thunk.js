import { setUserData, setDisconnectUser, setUserDataError } from "./actions";

// THUNK

export const login = (user) => async (dispatch) => {
  console.log("USER SENT TO THUNK ==>", user);
  //   try {
  //     return auth;
  //   } catch (err) {
  //     console.error("AUTHENTICATE_ERROR", err);
  //     dispatch(setUserDataError(err.toString()));
  //   }
};
