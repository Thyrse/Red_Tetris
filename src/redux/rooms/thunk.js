export const hydrateStore = async (dispatch) => {
  // console.log("HYDRATESTORE");
  try {
    // fill store
    await Promise.all([
      dispatch(fetchWorkspaces()),
      dispatch(fetchChatrooms()),
    ]);
  } catch (e) {
    console.error("HYDRATESTORE_ERROR", e);
  }
};
