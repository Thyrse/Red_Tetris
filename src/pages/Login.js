/* eslint-disable array-callback-return */
// REACT
import React, { useState, useEffect, useContext, useCallback } from "react";
// REACT ROUTER
import { Link, useHistory } from "react-router-dom";
import socketIOClient from "socket.io-client";
import { useDispatch, useSelector } from "react-redux";
import { setUserData } from "../redux/auth/actions";

/**
 * Component that displays the patient page,
 * contains the list of patients,
 * as well as the components for display and editing,
 * drawers and modal
 */
const Login = () => {
  const socket = socketIOClient.connect("http://localhost:4000");
  const [username, setUsername] = useState();
  const history = useHistory();
  const dispatch = useDispatch();

  const handleChange = (e) => {
    setUsername(e.target.value);
  };
  console.log("SOCKET ==>", socket);
  const handleSubmit = (e) => {
    e.preventDefault();
    socket.emit("login", username.toUpperCase());
    dispatch(
      setUserData({ username: username.toUpperCase(), socketID: socket.id })
    );
    // dispatch(setUserData(username.toUpperCase()));
    history.push("/home");
  };

  console.log("USERNAME ==>", username);
  return (
    <>
      <div className="d-flex justify-content-center">
        <div className="login p-3 shadow rounded text-center text-white">
          <div>
            <h3>Connect to play!</h3>
          </div>
          <form onSubmit={handleSubmit}>
            <div className="m-3">
              <label>Username:</label>
              <input
                type="text"
                name="username"
                id="username"
                className="form-control"
                placeholder="Tefourge, Scao..."
                onChange={(e) => handleChange(e)}
              />
            </div>
            <div className="d-flex justify-content-center">
              <button type="submit" className="btn btn-white">
                Play
              </button>
            </div>
          </form>
        </div>
      </div>
    </>
  );
};

export default Login;
