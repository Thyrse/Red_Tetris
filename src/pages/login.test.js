import React from "react";
import { mount, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Login from "./Login";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import socketIOClient from "socket.io-client";

const socket = socketIOClient.connect("http://localhost:4000");

console.log("SOCKET IN UNIT TEST ==>", socket);

const middlewares = [thunk];

const mockJest = jest.fn();

const mockStore = configureStore(middlewares);
const initialState = {
  userDatas: null,
  roomsList: [],
  usersList: null,
};

const store = mockStore(initialState);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockJest,
  }),
}));

const props = {
  socket: socket,
};

it("Login rendering upon arrival", () => {
  const wrapper = renderer
    .create(
      <Provider store={store}>
        <Login {...props} />
      </Provider>
    )
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});

it("Should throw an error on submission without username provided", () => {
  // const fakeEvent = { preventDefault: () => console.log("preventDefault") };
  const wrapper = mount(
    <Provider store={store}>
      <Login {...props} />
    </Provider>
  );
  // renderer
  //   .create(
  //     <Provider store={store}>
  //       <Login />
  //     </Provider>
  //   )
  //   .toJSON();
  const username = "Ziphlot";
  const event = { target: { value: username } };
  expect(wrapper.find(".form-control").simulate("change", event));
  expect(wrapper.find(".form-control").props().value).toBe(username);

  // wrapper.find(".form-login").simulate("submit", fakeEvent);
  // console.log(wrapper.find(".form-control").name);
  // expect(wrapper.find(".form-control").name).toBeTruthy();
});

it("Should redirect on form submission if an username is provided", () => {
  const fakeEvent = { preventDefault: () => console.log("preventDefault") };
  const wrapper = mount(
    <Provider store={store}>
      <Login {...props} />
    </Provider>
  );
  // renderer
  //   .create(
  //     <Provider store={store}>
  //       <Login />
  //     </Provider>
  //   )
  //   .toJSON();
  const username = "Ziphlot";
  const event = { target: { value: username } };
  expect(wrapper.find(".form-control").simulate("change", event));
  wrapper.find(".form-login").simulate("submit", fakeEvent);
  expect(mockJest).toHaveBeenCalledWith("/home");

  // wrapper.find(".form-login").simulate("submit", fakeEvent);
  // console.log(wrapper.find(".form-control").name);
  // expect(wrapper.find(".form-control").name).toBeTruthy();
});

// it("Login rendering upon arrival ENZYME", () => {
//   const tree = mount(<Provider store={store} />);
//   expect(toJson(tree)).toMatchSnapshot();
// });
// describe("<Login />", () => {
//   test("Login rendering upon arrival", () => {
//     const wrapper = renderer.create(<Provider store={store} />).toJSON()
//     expect(wrapper).toMatchSnapshot();
//   });

// test("Should throw an error on submission without username provided", () => {
//   const fakeEvent = { preventDefault: () => console.log("preventDefault") };
//   const wrapper = shallow(
//     <Provider store={store}>
//       <Login />
//     </Provider>
//   );
//   expect(wrapper.find(".form-login").length).toBe(1);
//   wrapper.find(".form-login").simulate("submit", fakeEvent);
//   expect(wrapper.find(Notification).length).toBe(1);

// const username = "Ziphlot";
// const event = { target: { value: username } };

// wrapper.find(StyledTextField).simulate("change", event);
// expect(wrapper.find(StyledTextField).props().value).toBe(username);
// expect(wrapper.find(StyledTextField).props().error).not.toBeTruthy();
// wrapper.find(StyledForm).simulate("submit", {
//   preventDefault: () => {},
// });
// expect(props.connectPlayer.mock.calls.length).toBe(1);
// });
// });
