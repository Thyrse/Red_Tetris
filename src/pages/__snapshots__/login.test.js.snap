// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login rendering upon arrival 1`] = `
<div
  className="d-flex justify-content-center"
>
  <div
    className="login p-3 shadow rounded text-center text-white"
  >
    <div>
      <h3>
        Connect to play!
      </h3>
    </div>
    <form
      className="form-login"
      onSubmit={[Function]}
    >
      <div
        className="m-3"
      >
        <label>
          Username:
        </label>
        <input
          className="form-control"
          error={0}
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Thyrse, Ziphlot..."
          type="text"
          value=""
        />
      </div>
      <div
        className="d-flex justify-content-center"
      >
        <span
          aria-describedby={null}
          className="p-2"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Your username must contain between 1 and 10 alphanumeric characters."
        >
          <button
            className="btn btn-white"
            disabled={true}
            type="submit"
          >
            Play
          </button>
        </span>
      </div>
    </form>
  </div>
</div>
`;
